src/aws/auth.rs

```diff
@@ -208,8 +221,10 @@ impl AwsAuthentication {
                 ));
                 if let Some(assume_role) = assume_role {
                     let auth_region = region.clone().map(Region::new).unwrap_or(service_region);
+                    let auth_external_id = external_id.clone().unwrap();
```

Comment:
This ended up causing a regression in https://github.com/vectordotdev/vector/issues/18340 as the `external_id` is not a required field.